Use upstream apt module, expand Ubuntu support - #4

initrd opened the pull request:
Fixes for #4:

Modify gpg key import methods. Couldn't get tests to pass with apt::key,
had to use an id + URL instead.

initrd committed:
Use upstream apt module, expand Ubuntu support - #4

Modify gpg key import methods. Couldn't get tests to pass with apt::key,
had to use an id + URL instead.

gcoxmoz commented on manifests/generic.pp:
```diff
@@ -18,8 +18,7 @@
       # Updated 2020-03, see https://help.duo.com/s/article/5503
-      source => 'puppet:///modules/duo_unix/DUO-GPG-PUBLIC-KEY',
-      notify => Exec['Duo Security GPG Import'];
+      source => "puppet:///modules/${module_name}/DUO-GPG-PUBLIC-KEY",
```
I'd like to recommend not doing the ${module_name} thing.
It makes sense for what we were doing internally with module shuffling in our own repos, but there are a good number of places where you already have assumed the name of the module.  So the right answer would be "either do it everywhere, or nowhere", and style-wise / people don't do it.  So I'd like to reject that line.

initrd replied:
SGTM. Replaced all occurrences of `${module_name}` with `duo_unix`.

gcoxmoz commented on manifests/apt.pp:
```diff
@@ -13,31 +13,19 @@
-    command     => '/usr/bin/apt-get update',
-    refreshonly => true
-  }
+  if $::duo_unix::manage_repo {
```
I don't disagree with this line, but I also can't find any uses of 'manage_repo' that are about managing apt sources.  So I think there's a slight bit of surprise to this behavior.  This, IMO, warrants a comment against the manage_repo param.

initrd replied:
SGTM, replaced all occurrences `${module_name}` with `duo`.

gcoxmoz commented on manifests/apt.pp:
```diff
@@ -13,31 +13,19 @@
+      location => $repo_uri,
+      repos    => 'main',
+      key      => {
+        'id'     => '08C2A645DDF240B85844068D7A450864C1A07A85',
```
"future-me who is going to stumble over this after the cert expired and we're rushing to get a new release pushed" would love the command to calculate/verify that fingerprint. :)

initrd replied:
This fingerprint was just me importing the key into gpg and running `gpg --list-fingerprints`. There might be a better way to do this using `apt-key`, but I'm not familiar enough with the tool.

gcoxmoz commented on metadata.json:
```diff
@@ -52,7 +52,9 @@
       "operatingsystem": "Ubuntu",
       "operatingsystemrelease": [
-        "14.04"
+        "14.04",
```
Drop 14?
Also: metadata.json isn't calling for apt as a requirement: deliberate?  And, any interest in adding Debian to the mix instead of just Ubuntu?

initrd replied:
Removed 14.04, added Debian 9.

initrd replied:
Added apt as dependency, had missed staging the change.

gcoxmoz commented on spec/classes/duo_unix_spec.rb:
```diff
@@ -35,7 +37,6 @@
           when 'RedHat'
             it { is_expected.to contain_class('duo_unix::yum') }
             it { is_expected.to contain_file('/etc/pki/rpm-gpg/DUO-GPG-PUBLIC-KEY') }
-            it { is_expected.to contain_yumrepo('duosecurity') }
```
Why drop the yumrepo test?  Seems like it'd still be useful.

initrd replied:
The next line [duplicates](https://github.com/mozilla-it/puppet-duo_unix/blob/master/spec/classes/duo_unix_spec.rb#L39) the check.

gcoxmoz replied:
clearly I excel at reading.  :P

initrd committed:
- Replace ${module_name} with actual module name
- Add puppetlabs/apt as dependency, update debian/ubuntu support
- Add comment about expanded manage_repo functionality
